include/pops/network.hpp

```diff
@@ -731,15 +813,26 @@ class Network
                 stream.seekg(starting_position);
                 break;
             }
-            if (label == "cost") {
+            if (label == "probability") {
+                if (has_cost) {
+                    std::runtime_error(
+                        "The cost column must be after the probability column");
+                }
```

Comment:
I wonder if this is not redundant when you test for column number below anyway.

Reply:
Yes, captured by the test below, too, but provides more a relevant message to the user. I added comment exampling that. Also added tests and discovered a missing throw there (we don't have automatic checks implemented to catch that). Done in c8c9949.